File: app/src/main/java/com/june/mvpkj/j/login/LoginActivity.java
package com.june.mvpkj.j.login;

import android.view.View;
import android.widget.Toast;

import com.june.mvpkj.databinding.ActivityLoginJBinding;
import com.june.mvpkj.j.base.BaseActivity;

public class LoginActivity extends BaseActivity<LoginPresenter> implements LoginContract.View{

    private ActivityLoginJBinding binding;

    @Override
    protected LoginPresenter createPresenter() {
        return new LoginPresenter(this);
    }

    @Override
    protected View layout() {
        binding = ActivityLoginJBinding.inflate(getLayoutInflater());
        return binding.getRoot();
    }

    @Override
    protected void init() {

    }

    @Override
    public void invalidLogin(String error) {
        Toast.makeText(this,error,Toast.LENGTH_SHORT).show();
    }

    @Override
    public void successResponse() {

    }

    @Override
    public void errorResponse() {

    }
}

File: app/src/main/java/com/june/mvpkj/k/login/LoginActivity.kt
package com.june.mvpkj.k.login

import android.content.Intent
import android.view.View
import android.widget.Toast
import com.june.mvpkj.R
import com.june.mvpkj.databinding.ActivityLoginBinding
import com.june.mvpkj.k.base.BaseActivity
import com.june.mvpkj.k.main.MainActivity

class LoginActivity : BaseActivity<LoginContract.View,LoginPresenter>(),LoginContract.View,
    View.OnClickListener {

    private val binding: ActivityLoginBinding by lazy { ActivityLoginBinding.inflate(layoutInflater) }

    override fun createPresenter(): LoginPresenter {
        return LoginPresenter(this)
    }

    override fun layout(): View {
        return binding.root
    }

    override fun init() {
        binding.loginTv.setOnClickListener(this)
    }

    override fun invalidLogin(error: String) {
        Toast.makeText(this,error,Toast.LENGTH_SHORT).show()
    }

    override fun successResponse() {
        val intent = Intent(this,MainActivity::class.java)
        startActivity(intent)
    }

    override fun errorResponse() {
        TODO("Not yet implemented")
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.login_tv -> {
                presenter?.login(binding.usernameEt.text.toString(),binding.passwordEt.text.toString())
            }
        }
    }
}